dll/win32/shell32/shlfileop.cpp

```diff
@@ -1219,6 +1219,34 @@ static void destroy_file_list(FILE_LIST *flList)
     HeapFree(GetProcessHeap(), 0, flList->feFiles);
 }
 
+static CStringW try_find_new_name(LPCWSTR szDestPath)
+{
+    CStringW mask(szDestPath);
+    CStringW ext(PathFindExtensionW(szDestPath));
+
+    // cut off extension before inserting a "new file" mask
+    if (!ext.IsEmpty())
+    {
+        mask = CStringW(mask.Left(mask.GetLength() - ext.GetLength()));
```

Comment:
CString has `operator+=`, `operator=` and `operator+` defined. So you could write it as follows: